pallets/ethereum-checked/src/tests.rs

```diff
@@ -0,0 +1,138 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(test)]
+
+use super::*;
+use mock::*;
+
+use ethereum::{ReceiptV3, TransactionAction, TransactionV2 as Transaction};
+use frame_support::{assert_noop, assert_ok};
+use sp_runtime::DispatchError;
+
+#[test]
+fn transact_works() {
+    ExtBuilder::default().build().execute_with(|| {
+        let store_tx = CheckedEthereumTx {
+            gas_limit: U256::from(1_000_000),
+            action: TransactionAction::Call(contract_address()),
+            value: U256::zero(),
+            // Calling `store(3)`
+            input: hex::decode(
+                "6057361d0000000000000000000000000000000000000000000000000000000000000003",
+            )
+            .unwrap(),
+            maybe_access_list: None,
+        };
+        assert_ok!(EthereumChecked::transact(
+            RawOrigin::XcmEthereumTx(ALICE).into(),
+            store_tx.clone()
+        ));
+        assert_ok!(EthereumChecked::transact(
+            RawOrigin::XcmEthereumTx(ALICE).into(),
+            store_tx
+        ));
+        let pending = pallet_ethereum::Pending::<TestRuntime>::get();
```

Comment:
Breaking changes in Frontier v0.9.40 makes `Pending` not public anymore. I'll make a PR to our fork later to revert it back to public.